clang/test/Analysis/cxx23-static-operator.cpp

```diff
@@ -0,0 +1,36 @@
+// RUN: %clang_analyze_cc1 -std=c++2b -verify %s \
+// RUN:   -analyzer-checker=core,debug.ExprInspection
+
+template <typename T> void clang_analyzer_dump(T);
+
+struct Adder {
+  int data;
+  static int operator()(int x, int y) {
+    clang_analyzer_dump(x); // expected-warning {{1}}
+    clang_analyzer_dump(y); // expected-warning {{2}}
+    return x + y;
+  }
+};
+
+void static_operator_call_inlines() {
+  Adder s{10};
+  clang_analyzer_dump(s(1, 2)); // expected-warning {{3}}
+}
+
+struct DataWithCtor {
+  int x;
+  int y;
+  DataWithCtor(int v) : x(v + 10), y(v + 20) {}
```

Comment:
Could we also dump `this` this to confirm that it is param var regions for `operator[]` and we trigger copy elsion?

Reply:
It doesn't look like it. I was expecting `&s`, but it's `&v`.

Reply:
It seem correct to me. With guaranteed copy elision, this object is constructed directly into parameter `DataWithCtor v` of `operator[]`, so we get `&v` as memRegion to object under construction.